cadl-tests/src/main/java/com/cadl/paged/PagedAsyncClient.java

```diff
@@ -62,4 +63,22 @@ public final class PagedAsyncClient {
     public PagedFlux<BinaryData> list(RequestOptions requestOptions) {
         return this.serviceClient.listAsync(requestOptions);
     }
+
+    /**
+     * List Resource resources.
+     *
+     * @throws com.azure.core.exception.HttpResponseException thrown if the request is rejected by server.
+     * @throws ClientAuthenticationException thrown if the request is rejected by server on status code 401.
+     * @throws ResourceNotFoundException thrown if the request is rejected by server on status code 404.
+     * @throws ResourceModifiedException thrown if the request is rejected by server on status code 409.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return paged collection of Resource items as paginated response with {@link PagedFlux}.
+     */
+    @Generated
+    @ServiceMethod(returns = ReturnType.COLLECTION)
+    public PagedFlux<Resource> list() {
+        // Generated convenience method for list
+        RequestOptions requestOptions = new RequestOptions();
+        return list(requestOptions).mapPage(protocolMethodData -> protocolMethodData.toObject(Resource.class));
```

Comment:
`mapPage` is deprecated in core, we might need to switch to some util code later.